### PR TITLE
Add model wallet and associated with user set initial balance after c…

### LINKED ISSUE #4: Implement a user wallet
# REQUIREMENTS
Create a model and a table in postgresql

### PR DESCRIPTION
**Pull request by alexs**
close issue #4

Thanks 🙌🏻

Co-authored-by: AlejandroCuevasZepeda <AlejandroCuevasZepeda@users.noreply.github.com>
Co-authored-by: alanRazoMen <alanRazoMen@users.noreply.github.com>
Co-authored-by: MrCano369x <MrCano369x@users.noreply.github.com>

# Description
What did you implemented/Why did you implemented this:
 - We added model Wallet to save current coins by user 
 
 Examples:
 - What?: I've added model to know the current user coins in its wallet. See #4 for more information.

The first approval of this pr will earn 50 BrightCoins extra 💸

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,4 +3,11 @@ class User < ApplicationRecord
...
          :recoverable, :rememberable, :validatable
-end
+
+  has_one :wallet
+  after_create :create_wallet
+
+  def create_wallet
+    build_wallet(coins: 50.00).save
+  end
+end
```
Comment (YilKanda) [inline app/models/user.rb]: Add a blank line

File: app/views/home/index.html.erb [hunk 2]
```diff
@@ -1,2 +1,3 @@
-<h1>Bienvenido</h1>
-<p>Juega ya!</p>
+<h1>Welcome</h1>
+<p>Play now!</p>
+<h2>Your balance is: <%= number_to_currency(current_user.wallet.coins) %> BrightCoins</h2>
```
Comment (YilKanda) [inline app/views/home/index.html.erb]: Add a blank line

File: config/database.yml [hunk 3]
```diff
@@ -24,8 +24,8 @@ default: &default
 development:
   <<: *default
   database: proyecto_final_pf_f22_t3_development
-  username: 
-  password: 
+  username: <%= Rails.application.credentials.database.username %>
+  password: <%= Rails.application.credentials.database.password %>
 
   # The specified database role being used to connect to postgres.
   # To create additional roles in postgres see `$ createuser --help`.
```
Comment (YilKanda) [inline config/database.yml]: This file should not appear on GitHub, instead there should be an example file.

File: test/models/wallet_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class WalletTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (YilKanda) [inline test/models/wallet_test.rb]: Code that you do not use, delete it.

**Review by YilKanda — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (alexs) [reply app/models/user.rb]: Ok, done! thx!

File: app/views/home/index.html.erb (hunk 2 above)
Comment (alexs) [reply app/views/home/index.html.erb]: Ok, done! thx!

File: config/database.yml (hunk 3 above)
Comment (alexs) [reply config/database.yml]: Ok, removed file thx!

File: test/models/wallet_test.rb (hunk 4 above)
Comment (alexs) [reply test/models/wallet_test.rb]: Ok, removed files thx!

**Review by YilKanda — DISMISSED** [review]
LGTM!

**Review by dannharo — CHANGES_REQUESTED** [review]
Good job but.. you always have to include tests on your code, even if there is no requirement for that, define if you are going to use `minitest` or `rspec`, so that, create specs for the new model (relations test at least).

**Comment by alexs** [comment]
> Good job but.. you always have to include tests on your code, even if there is no requirement for that, define if you are going to use `minitest` or `rspec`, so that, create specs for the new model (relations test at least).

@dannharo we already solved this

**Review by dannharo — APPROVED** [review]

**Review by Frcomparan — APPROVED** [review]
LGTM

**Review by MrCano369x — APPROVED** [review]
se ve chido

**Pull request merged**